samples/net/rpl_border_router/src/rpl.c

```diff
@@ -4,8 +4,10 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#define LOG_MODULE_NAME net_rpl_br_rpl
-#define NET_LOG_LEVEL LOG_LEVEL_DBG
+#define LOG_LEVEL LOG_LEVEL_DBG
+
+#include <logging/log.h>
+LOG_MODULE_DECLARE(net_rpl_br_sample, LOG_LEVEL);
```

Comment:
Ok.

I've been looking a bit into how to get rid of those `if (LEVEL>=DBG)` since ideally compiler should take care of that. An option would be to mark helper function (like `net_addr_ntop`) as pure (`__attribute__ ((pure))`. That indicates that function has no effects except the return value. When function is marked that way then compiler will cut it once it is noticed that return value is not used (because LOG_DBG is cut by compiler). However, to make this happen you need to use return value. Example from`rpl.c`:
```
char me_mem[NET_IPV6_ADDR_LEN];
char other_mem[NET_IPV6_ADDR_LEN];
char * me, other;

me = net_addr_ntop(AF_INET6, &dio->dag_id, other_mem, NET_IPV6_ADDR_LEN);
other = net_addr_ntop(AF_INET6, &rpl.dag_id, me_mem, NET_IPV6_ADDR_LEN);

LOG_DBG(..., me, other);
```

what's your opinion?

Reply:
One note: if attribute is added to `net_addr_ntop` then we need to ensure that return value is always used across whole zephyr because compiler will cut function if return value is not used.

Reply:
> what's your opinion?

Something like that could be done but not for this patch. Marking functions as pure would also need some extra review about how the values are used as you mention.

Reply:
sure, i agree that it's not in the scope of that patch